module/input/SpeechIntent/src/SpeechIntent.cpp

```diff
@@ -0,0 +1,405 @@
+                return;
+            }
+            buffer[bytes_read] = 0;
+
+            if (output_enabled) {
+                char* intent_json = voice2json_recognize_intent(buffer);
+                if (!intent_json) {
+                    NUClear::log<NUClear::FATAL>(
+                        fmt::format("({}:{}) failed to recognize intent, errno = {}", __FILE__, __LINE__, errno));
+                    return;
+                }
+
+                std::unique_ptr<SpeechIntentMsg> intent = parse_voice2json_json(intent_json, strlen(intent_json));
+                emit(std::move(intent));
+
+                free(intent_json);
+            }
+        });
+
+        on<IO>(voice2json_proc.stderr, IO::READ).then([this] {
+            char buffer[0x1000];
+            ssize_t bytes_read = read(voice2json_proc.stderr, buffer, sizeof(buffer) - 1);
+            if (bytes_read == -1) {
+                NUClear::log<NUClear::FATAL>(fmt::format("({}:{}) Failed to read from stderr, errno = {}",
+                                                         __FILE__,
+                                                         __LINE__,
+                                                         errno));
+                return;
+            }
+            buffer[bytes_read] = 0;
+            NUClear::log<NUClear::FATAL>(fmt::format("({}:{})\nVOICE2JSON stderr, errno = {} ({} bytes)\n{}",
+                                                         __FILE__,
+                                                         __LINE__,
+                                                         errno,
+                                                         bytes_read,
+                                                         buffer));
+        });
+
+
+        on<Trigger<SpeechInputSetOutputMsg>>().then(
+            [this](const SpeechInputSetOutputMsg& msg) { output_enabled = msg.enabled; });
+
+        on<Trigger<SpeechInputRecognizeWavFile>>().then([this](const SpeechInputRecognizeWavFile& msg) {
+            if (!write_audio_to_file(voice2json_proc.stdin, (char*) msg.filename.data())) {
+                NUClear::log<NUClear::FATAL>(
+                    fmt::format("({}:{}) Failed to write to {}", __FILE__, __LINE__, msg.filename.data()));
+            }
+        });
+    }
+
+    void SpeechIntent::recognize_wav(std::string wav_filename) {
+        std::unique_ptr<SpeechInputRecognizeWavFile> msg =
+            std::make_unique<SpeechInputRecognizeWavFile>(SpeechInputRecognizeWavFile{wav_filename});
+        emit(std::move(msg));
+    }
+
+    SpeechIntent::SpeechIntent(std::unique_ptr<NUClear::Environment> environment)
+        : Reactor(std::move(environment)), config{} {
+
+
+        on<Configuration>("SpeechIntent.yaml").then([this](const Configuration& cfg) {
+            this->log_level = cfg["log_level"].as<NUClear::LogLevel>();
+        });
+
+        on<Trigger<CommandLineArguments>>().then([this](const CommandLineArguments& args) {
```

Comment:
Our reasoning for splitting it out - We don't want that code to be re-run in the case of the configuration file being reloaded.

Reply:
@Bidski Would this be better handled as this? If you want to make sure it doesn't run again if the config file is changed but want it to run straight away?
handle = on<config> {
Do things
handle.disable();
}

Reply:
Using command line arguments in this way seems a little pointless tbh. I would just move all of this logic into `on<Configuration>` and not use `CommandLineArguments` at all for this.

Reply:
Why do you think it's pointless?

The reason why we're trying to access command line argument is so that we have some way of testing the module with different audio sources. In particular, in a container environment, we can't access the microphone, so the audio source is instead a wav file. Are you recommending that we hard code a path into the configuration file that points to a wav file?

Reply:
Its pointless because you can achieve all of the same functionality using the configuration file.
You could easily set up a `sound_source` entry in the config file, and if that represents a valid file (you can use `std::filesystem` for this) then you load that as an audio file, if `sound_source` is `mic` (or some other appropriate keyword) then you look for the appropriate hardware and get your audio data from there.

Also, are you aware of [this](https://stackoverflow.com/a/48795232/4795763)? Not sure if this would work in WSL.